master/internal/core.go

```diff
@@ -445,7 +446,8 @@ func (m *Master) tryRestoreExperiment(sema chan struct{}, e *model.Experiment) {
 	defer func() { <-sema }()
 	// Check if the returned config is the zero value, i.e. the config could not be parsed
 	// correctly. If the config could not be parsed, mark the experiment as errored.
-	if !reflect.DeepEqual(e.Config, model.ExperimentConfig{}) {
+	// XXX: is this even slightly right??
```

Comment:
cool.  I remember that it used to have to do with the unmarshaling logic; in error cases it would return an empty config for some reason.

I agree that that should no longer be the case.